packages/shorebird_cli/lib/src/third_party/flutter_tools/lib/src/base/io.dart

```diff
@@ -79,7 +84,7 @@ bool _inUnitTest() {
 void setExitFunctionForTests([ExitFunction? exitFunction]) {
   _exitFunction = exitFunction ??
       (int exitCode) {
-        throw ProcessExit(exitCode, immediate: true);
+        throw const UnexpectExitCalled();
```

Comment:
What is `UnexpectExitCalled()` used for? Seems like we should just be able to throw a `StateError` instead if we never want this to be hit.

Reply:
Yeah, `UnsupportedError` fits nicely too, I forgot about that one